compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/ModuleLevelVariableDeclarationAnalysisTask.java

```diff
@@ -0,0 +1,84 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.graphql.compiler;
+
+import io.ballerina.compiler.api.symbols.VariableSymbol;
+import io.ballerina.compiler.syntax.tree.ModuleVariableDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ObjectConstructorExpressionNode;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.stdlib.graphql.compiler.schema.generator.GeneratorUtils;
+import io.ballerina.stdlib.graphql.compiler.schema.generator.GraphqlModifierContext;
+import io.ballerina.stdlib.graphql.compiler.schema.types.Schema;
+import io.ballerina.stdlib.graphql.compiler.service.InterfaceFinder;
+import io.ballerina.stdlib.graphql.compiler.service.validator.ServiceValidator;
+
+import java.util.Map;
+
+import static io.ballerina.stdlib.graphql.compiler.Utils.hasCompilationErrors;
+import static io.ballerina.stdlib.graphql.compiler.Utils.isGraphQLServiceObjectDeclaration;
+
+/**
+ * Validates a Ballerina GraphQL Service variable declaration.
+ */
+public class ModuleLevelVariableDeclarationAnalysisTask extends ServiceAnalysisTask {
+
+    public ModuleLevelVariableDeclarationAnalysisTask(Map<DocumentId, GraphqlModifierContext> nodeMap) {
+        super(nodeMap);
+    }
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        if (hasCompilationErrors(context)) {
+            return;
+        }
+
+        ModuleVariableDeclarationNode moduleVariableDeclarationNode = (ModuleVariableDeclarationNode) context.node();
+        if (!isGraphQLServiceObjectDeclaration(moduleVariableDeclarationNode)) {
+            return;
+        }
+
+        if (moduleVariableDeclarationNode.initializer().isEmpty()) {
+            return;
+        }
+        ObjectConstructorExpressionNode graphqlServiceObjectNode
+                = (ObjectConstructorExpressionNode) moduleVariableDeclarationNode.initializer().get();
+        InterfaceFinder interfaceFinder = getInterfaceFinder(context);
+        ServiceValidator serviceObjectValidator = getServiceValidator(context, graphqlServiceObjectNode,
+                                                                      interfaceFinder);
+        if (serviceObjectValidator.isErrorOccurred()) {
+            return;
+        }
+
+        String description = getDescription(context, moduleVariableDeclarationNode);
+        Schema schema = generateSchema(context, interfaceFinder, graphqlServiceObjectNode, description);
+        DocumentId documentId = context.documentId();
+        addToModifierContextMap(documentId, moduleVariableDeclarationNode, schema);
```

Comment:
Can't we remove the duplication here with the abstract classes? Since the analysis tasks behave similarly, we can pass the corresponding analysis task and then check for errors, find interfaces, generate schema, etc.